### ISSUE TITLE
[Enhancement]: aws_rds_cluster doesn't expose publicly_accessible

### ISSUE THREAD

**Issue by colemickens** [hashicorp/terraform-provider-aws#39618]
### Description

AWS makes it confusing enough to understand clusters and their public accessibility.

1. If you don't specify it via the API, it is implicitly determined based on if the VPC has a IG.
2. The docs say NOTHING about changing the public accessibility of cluster endpoints after creation.
3. The dashboard just... doesn't show the same option for public access after creation

So:
* I have no idea if a cluster can retro-actively be made publicly-accessible. It seems like quite an odd limitaiton if not.
* I can't... try it and see, because the AWS Terraform module doesn't expose this option for the cluster, only for cluster instances.

Ask:
* Someone to check my understanding - why does AWS make this so opaque and hard to change after creation?
* Please support `publicly_available` for the cluster, so I can set/change it as needed.

### Affected Resource(s) and/or Data Source(s)

* aws_rds_cluster

### Would you like to implement a fix?

None

**Comment by colemickens** [comment]
To further elaborate on my confusion, ModifyClusterRequest doesn't let you change PubliclyAvailable, yet I can make the individual under-lying instances public still?

**Comment by colemickens** [comment]
Actually, much more concerningly, this seems to mean there's no way to **preclude public access** if you happen to be creating the RDS cluster attached to a VPC that happens to have an IG enabled...

**Comment by colemickens** [comment]
Oh, the cluster endpoint appears to CNAME to the instance anyway. Why oh why can't AWS just document things like this?

**Comment by dreid** [comment]
> * If you don't specify it via the API, it is implicitly determined based on if the VPC has a IG.

I do not believe the AWS docs are correct here, (if they are then the configuration is more nuanced than the AWS docs say) afaict it is only possible to get a publicly accessible multi-az rds cluster by specifying `PubliclyAccessible` at create time.

I can confirm that it does not appear to be possible to specify this at modify time, and at least when you have a Multi-AZ/non-aurora cluster it is not possible to directly modify the db instances.

**Comment by sharpone74** [comment]
Is there any update on this?

**Comment by ewbankkit** [comment]
```go
	// Specifies whether the DB cluster is publicly accessible.
	//
	// Valid for Cluster Type: Multi-AZ DB clusters only
	//
	// When the DB cluster is publicly accessible and you connect from outside of the
	// DB cluster's virtual private cloud (VPC), its domain name system (DNS) endpoint
	// resolves to the public IP address. When you connect from within the same VPC as
	// the DB cluster, the endpoint resolves to the private IP address. Access to the
	// DB cluster is controlled by its security group settings.
	//
	// When the DB cluster isn't publicly accessible, it is an internal DB cluster
	// with a DNS name that resolves to a private IP address.
	//
	// The default behavior when PubliclyAccessible is not specified depends on
	// whether a DBSubnetGroup is specified.
	//
	// If DBSubnetGroup isn't specified, PubliclyAccessible defaults to true .
	//
	// If DBSubnetGroup is specified, PubliclyAccessible defaults to false unless the
	// value of DBSubnetGroup is default , in which case PubliclyAccessible defaults
	// to true .
	//
	// If PubliclyAccessible is true and the VPC that the DBSubnetGroup is in doesn't
	// have an internet gateway attached to it, Amazon RDS returns an error.
	PubliclyAccessible *bool
```
